src/xrc/src/lib.rs

```diff
@@ -332,6 +336,36 @@ impl std::ops::Mul for QueriedExchangeRate {
     /// identical to the base asset of the second struct.
     #[allow(clippy::suspicious_arithmetic_impl)]
     fn mul(self, other_rate: Self) -> Self {
+        let median_rate = match (self.median_rate, other_rate.median_rate) {
+            (Some(own_rate), Some(other_rate)) => {
+                let product_rate = (own_rate as u128)
+                    .saturating_mul(other_rate as u128)
+                    .saturating_div(RATE_UNIT as u128);
+                if product_rate <= (RATE_UNIT * RATE_UNIT) as u128 {
+                    Some(product_rate as u64)
+                } else {
+                    None
+                }
+            }
+            (_, _) => None,
+        };
```

Comment:
There is a significant problem. The forex rates that we have been storing into the store will not have the `median_rate` field set after the upgrade basically invalidating them. I have two ideas:

1. We can update the values in the post_upgrade.
2. We can update the median rate on the fly when accessed.

Reply:
This is indeed a significant problem. We definitely have to solve this problem before merging this PR.

Reply:
The latest changes should fix this problem. Please take another look!